pulp_container/app/serializers.py

```diff
@@ -126,6 +127,9 @@ class BlobSerializer(SingleArtifactContentSerializer):
     """
 
     digest = serializers.CharField(help_text="sha256 of the Blob file")
+    artifact = SingleContentArtifactField(
```

Comment:
why this is needed?

Reply:
This was a trick part. `BlobSerializer` inherits from `SingleArtifactContentSerializer` in which an artifact is a required field. With this PR, an artifact for `Blob.data` is now optional (because depending on the size of the blob we would store it in the database or keep it as a file).

Removing the `required=False` constraint would work, but our bindings would fail when we store the config or layer in DB (no artifact would be created) with:
```
>           raise ValueError("Invalid value for `artifact`, must not be `None`")  # noqa: E501
E           ValueError: Invalid value for `artifact`, must not be `None`
```

Reply:
hum.... I think serializing artifacts with "null value" was not an issue before because we were allowing it already:
https://github.com/pulp/pulpcore/blob/6fc92a61785dae6522702f2844e00ced901f86e3/pulpcore/app/serializers/fields.py#L31

with the `required=False` we removed the constraint of artifacts being None (not defined!?).

Reply:
my concern is more about that we will be serializing artifact=null and end user won't know whether it is broken/missing or in DB

Reply:
could we instead look into how we can change binding to cope with optional artifact?